Formula/r/recyclarr.rb

```diff
@@ -0,0 +1,33 @@
+class Recyclarr < Formula
+  desc "Automatically sync TRaSH Guides to your Sonarr and Radarr instances"
+  homepage "https://recyclarr.dev"
+  version "7.4.0"
+  url "https://github.com/recyclarr/recyclarr.git", tag: "v#{version}"
+  license "MIT"
+
+  depends_on "dotnet@8" => :build
+  # recyclarr has a dependency on git, which according to brew audit is always available
+
+  def install
+    os = OS.mac? ? "osx" : OS.kernel_name.downcase
+    arch = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch.to_s
+
+    args = %W[
+      --configuration Release
+      --runtime #{os}-#{arch}
+      --output #{libexec}
+      --self-contained true
+      -p:PublishSingleFile=true
+      -p:EnableCompressionInSingleFile=true
+      -p:PublishReadyToRunComposite=true
+      -p:TieredCompilation=false
+    ]
+
+    system "dotnet", "publish", "src/Recyclarr.Cli", *args
+    bin.install_symlink libexec/"recyclarr"
+  end
+
+  test do
+    assert_match "v#{version}", shell_output("#{bin}/recyclarr --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate temasterath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
can you share the docs how the command is being used?

Reply:
Docs are on recyclarr.dev, but I am also the author. I suppose I could use the `config` subcommand but that won't really exercise more than `--version`. Just by doing `--version` we're verifying that the application can be set up properly, which is a fairly good smoke test IMO.

Reply:
Can you run an actual command and assert a failure? That's generally a good test for us.

---

> but I am also the author

We frown on authors submitting their own work unless it is very popular.
https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae:~:text=We%20frown%20on%20authors%20submitting%20their%20own%20work%20unless%20it%20is%20very%20popular.

Reply:
> We frown on authors submitting their own work unless it is very popular

I appreciate you pointing this out to me. I was already aware of this, however, but decided to pursue this anyway since:

- No third party has contributed this in years of waiting (and I did wait)
- I wanted to rely on an exception being made due to the other conditions being met (quoted below).

> - be maintained (i.e. the last release wasn’t ages ago, it works without patching on all Homebrew-supported OS versions and has no outstanding, unpatched security vulnerabilities)
> - be stable (e.g. not declared “unstable” or “beta” by upstream)
> - be known (e.g. GitHub repositories should have >=30 forks, >=30 watchers or >=75 stars)
> - be used
> - have a homepage